fenix/app/src/main/java/org/mozilla/fenix/home/ToolbarView.kt

```diff
@@ -38,6 +40,11 @@ class ToolbarView(
      * Setups the home screen toolbar.
      */
     fun build() {
+        if (IncompleteRedesignToolbarFeature(context.settings()).isEnabled) {
```

Comment:
the `updateLayout` function might be a better fit for setting the visibility of the buttons. also, when dealing with flags, it's better not to set them only to one value – it opens up a possibility for cached state. something like this would be safer
```
binding.menuButton.isVisible(!context.settings()).isEnabled)
```